test/query.test.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var Query = require('../lib/query');
+
+describe('Query', function() {
```

Comment:
I get the sense that `Query` is not a public API, so these tests may be a hindrance during refactoring. What do you think about verifying this behavior through `Table#select`?

Reply:
I am testing most things in `select` because I see at as likely being the public API. 

The handful of error states I'm addressing here are actually impossible to reach via select because select does its own validation and parameter filtering and error catching before passing the query directly to select. So its not possible to have 100% code coverage with out these unit tests. See `lib/table:49` Should I remove them and add ignore? Or add a note?

I could move the `validates cellFormat params` but that feels like an okay low level unit test to me.

Reply:
If `Query` is truly private, then any code paths that can't be reached in production are dead.

We could certainly merge the tests even if that's true. They're correct, after all. My only concern is that, in the absence of documentation, the presence of tests may lead others to believe that the code they exercise is important.

@kasra We believe that `Query` is private. Could you confirm?

Reply:
Yep `Query` is intended to be private, so would prefer to not add tests specifically for it. I think it would be reasonable to remove the extra error checking in the `Query` constructor if they're not reachable, but let's also update the comment for the `Query` constructor to document that it expects the caller to first validate `params` by calling `Query.validateParams`.

Reply:
I completely removed and added comments about all of the unreachable error conditions which allowed me to completely remove the `query.test.js` file.